src/pages/AddTask.tsx

```diff
@@ -102,7 +103,27 @@ const AddTask = (props: AddTaskProps) => {
             color: colors.filter((color) => color.selected)[0].color,
         }
 
-        addTask(task).catch((err) => { alert(JSON.stringify(err)) }).then(() => props.goBack());
+        let taskId: Database['public']['Tables']['Tasks']['Row']['task_id'];
+
+        try {
+            taskId = await addTask(task);
+            props.goBack();
+        } catch (err) {
+            console.log(err);
+            props.goBack();
+            return;
```

Comment:
Since you're calling `props.goBack()` on both the `try` and `catch`, I'd suggest using a `finally`